### PR TITLE
Implemented read_all blocks and reorganized storage tests

### PR DESCRIPTION
**Pull request by eadventurous**
### Description of the Change

1. implemented read_all
2. reorganized storage tests
3. turned tests with async functions into async tests

Right now read_all assumes that we read blocks starting from `0` and finish when we can't find the block with the next height, meaning there can be no gaps in the sequence. This is an intended behavior.

### Benefits

Read all - implemented

### Possible Drawbacks 

What should we do when the sequence on disk is corrupted?

### Alternate Designs *[optional]*

Read all files in a directory and then check and reorder them.

Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### CODE REVIEW

File: src/storage/kura.rs [hunk 1]
```diff
@@ -309,68 +309,86 @@ impl Disk {
         Ok(model::Block::from(buffer))
     }
 
-    //TODO: implement reading all blocks
-    fn read_vec(&self) -> Vec<model::Block> {
-        Vec::new()
+    /// returns a sorted vector of blocks starting from 0 height to the top block
```
Comment (humb1t) [inline src/storage/kura.rs]: Docs like a normal sentences should start from the capital letter and end with full stop.

File: src/storage/kura.rs [hunk 2]
```diff
@@ -309,68 +309,86 @@ impl Disk {
...
+    async fn read_all(&self) -> Vec<model::Block> {
+        let mut height = 0;
+        let mut blocks = Vec::new();
+
```
Comment (humb1t) [inline src/storage/kura.rs]: no spaces needed

File: src/storage/kura.rs [hunk 3]
```diff
@@ -309,68 +309,86 @@ impl Disk {
...
+        while let Ok(block) = self.read(height).await {
+            blocks.push(block);
+            height += 1;
+        }
+
```
Comment (humb1t) [inline src/storage/kura.rs]: the same

File: src/storage/kura.rs [hunk 4]
```diff
@@ -309,68 +309,86 @@ impl Disk {
...
 
-    //TODO: implement reading all blocks
-    fn read_vec(&self) -> Vec<model::Block> {
-        Vec::new()
+    /// returns a sorted vector of blocks starting from 0 height to the top block
+    async fn read_all(&self) -> Vec<model::Block> {
+        let mut height = 0;
+        let mut blocks = Vec::new();
+
+        while let Ok(block) = self.read(height).await {
```
Comment (humb1t) [inline src/storage/kura.rs]: Maybe we can read all bytes at once?
https://docs.rs/async-std/1.5.0/async_std/io/trait.Read.html#method.read_to_end

Comment (eadventurous) [reply src/storage/kura.rs]: `Disk::read` function already internally reads all bytes at once for a given file, or you mean someting different?

Comment (humb1t) [reply src/storage/kura.rs]: Sry, forgot that we split blocks into one block one file.

**Pull request merged**